components/ca-context.js

```diff
@@ -0,0 +1,71 @@
+define(['document-register-element'], () => {
+    /**
+     * Web component context
+     */
+    class Context extends HTMLElement {
+
+        /** @returns {string} attribute id */
+        get id() {
+            return this.getAttribute('id');
+        }
+
+        /** @param {string} value, assigned to id attrinute */
+        set id(value) {
+            this.setAttribute('id', value);
+        }
+
+        /** @returns {string} attribute name */
+        get name() {
+            return this.getAttribute('name');
+        }
+
+        /** @param {string} value, assigned to name attrinute */
+        set name(value) {
+            this.setAttribute('name', value);
+        }
+
+        /** @returns {string} user path */
+        get value() {
+            return this.getUserPath();
+        }
+
+        /** @returns {string} type */
+        get type() {
+            return 'hidden';
+        }
+
+        /**
+         * Attempts to construct a path based on the users UI
+         * We use query selector over $ so we only get the first instance of any object
+         * @returns {string} path based values
+         */
+        getUserPath() {
+            const txtProp = ('textContent' in document.createElement('i')) ? 'textContent' : 'innerText';
```

Comment:
A terrible reason - in the original, the function is only invoked on the .value getter. This functionality isn't used anywhere else, so if anything, it should either move into the getter or become part of the class. It doesn't feel very much like a private function to me. Happy to roll it back, though.